cudax/include/cuda/experimental/detail/hierarchy_dimensions.cuh

```diff
@@ -0,0 +1,811 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX_DETAIL_HIERARCHY_DIMENSIONS
```

Comment:
We are actively moving away from "detail" folders.

The convention we are moving towards is that the content of header `cuda/meow` is within the folder `cuda/__meow`